config/config_test.go

```diff
@@ -105,5 +121,81 @@ func TestReadConfig(t *testing.T) {
 			}
 		})
 	}
+}
+
+func TestParseChainParams(t *testing.T) {
+	var parseChainParamTests = map[string]struct {
+		chainName   []string
+		chainParams chaincfg.Params
+	}{
+		"main network": {
+			[]string{"mainnet"},
+			chaincfg.MainNetParams,
+		},
+		"regtest": {
+			[]string{"regtest"},
+			chaincfg.RegressionNetParams,
+		},
+		"simnet": {
+			[]string{"simnet"},
+			chaincfg.SimNetParams,
+		},
+		"testnet3": {
+			[]string{"testnet3"},
+			chaincfg.TestNet3Params,
+		},
+		"undefined": {
+			[]string{},
+			chaincfg.MainNetParams,
+		},
+		"empty": {
+			[]string{""},
+			chaincfg.MainNetParams,
+		},
+	}
+	for testName, testData := range parseChainParamTests {
+		t.Run(testName, func(t *testing.T) {
+			// use a string builder and a single-value list to represent optionality
+			var b strings.Builder
+			fmt.Fprint(&b, "[Extensions.TBTC.BTCRefunds]")
+			for _, name := range testData.chainName {
+				fmt.Fprintf(&b, "\nChainName=\"%s\"", name)
+			}
+			config := &Config{}
+			if _, err := toml.Decode(b.String(), config); err != nil {
+				t.Fatal(err)
+			}
+			chainParams, err := config.Extensions.TBTC.BTCRefunds.ChainParams()
+			if err != nil {
+				t.Fatal(err)
+			}
+			if !reflect.DeepEqual(*chainParams, testData.chainParams) {
+				t.Errorf("unexpected net params\nexpected: %v\nactual:   %v", testData.chainParams, chainParams)
+			}
+		})
+	}
+}
+
+func TestParseChainParams_ExpectedFailure(t *testing.T) {
+	configString := fmt.Sprintf("[Extensions.TBTC.BTCRefunds]\nChainName=\"%s\"", "bleeble blabble")
+	config := &Config{}
+	if _, err := toml.Decode(configString, config); err != nil {
+		t.Fatal(err)
+	}
+	_, err := config.Extensions.TBTC.BTCRefunds.ChainParams()
+	expectedError := "unable to find chaincfg param for name: [bleeble blabble]"
+	if err == nil {
+		t.Fatalf("expecting an error but found none")
+	}
+	if !errorContains(err, expectedError) {
```

Comment:
Thoughts on comparing it here instead of defining a function?
```suggestion
	if expectedError != err.Error() {
```